apps/custom-ui-app-docusign/server/util/helper.js

```diff
@@ -0,0 +1,145 @@
+        debug(error);
+        throw new Error("Getting new token from refresh token failed");
+    }
+};
+
+// Gets the page context for rendering the HBS templates
+function getPageContext(company_id, item_id, details) {
+    return {
+        company_id,
+        item_id,
+        details
+    }
+}
+
+// Get the DocuSign templates via REST API
+async function getDocuSignTemplates(account_id, docusign_token) {
+    try {
+        const templates = await axios({
+            method: 'GET',
+            url: `https://demo.docusign.net/restapi/v2.1/accounts/${account_id}/templates`,
+            headers: {
+                'Authorization': `Bearer ${docusign_token}`
+            }
+        });
+        return templates.data;
+    } catch (error) {
+        debug(error);
+        throw new Error("API request for getting DocuSign templates failed");
+    }
+}
+
+// Get the DocuSign Template Preview URL via REST API
+async function generateTemplatePreviewURL(template_id, account_id, docusign_token) {
+    try {
+        return await axios({
+            method: 'POST',
+            url: `https://demo.docusign.net/restapi/v2.1/accounts/${account_id}/templates/${template_id}/views/edit`,
+            headers: {
+                'Authorization': `Bearer ${docusign_token}`
+            },
+            data: {
+                "returnUrl": "https://example.com"
+            },
+            json: true
+        });
+    } catch (error) {
+        debug(error);
+        throw new Error("API Request for generating template preview URL failed");
+    }
+}
+
+// Creates a document from DocuSign template
+async function createDocumentFromTemplate(account_id, template_id, docusign_token, signer_email, signer_name) {
+    try {
+        return await axios({
+            method: 'POST',
+            url: `https://demo.docusign.net/restapi/v2.1/accounts/${account_id}/envelopes`,
+            headers: {
+                'Authorization': `Bearer ${docusign_token}`
+            },
+            data: {
+                "templateId": template_id,
+                "templateRoles": [{
+                    "email": signer_email,
+                    "name": signer_name,
+                    "roleName": "signer"
+                }],
+                "status": "sent"
+            },
+            json: true
+        });
+    } catch (error) {
+        debug(error);
+        throw new Error("API request for sending DocuSign document to the recipient failed");
+    }
+}
+
+// Gets information about a particular deal (by ID) in Pipedrive
+async function getDeal(deal_id, access_token) {
+    try {
+        const deal = await axios({
+            method: 'GET',
+            url: `https://api.pipedrive.com/v1/deals/${deal_id}`,
+            headers: {
+                'Authorization': `Bearer ${access_token}`
+            }
+        });
+        return deal.data;
+    } catch (error) {
+        debug(error);
+        throw new Error("API request to get deal by ID failed");
+    }
```

Comment:
You could use our Pipedrive API client for this

Reply:
Taken care :)